### ISSUE TITLE
Models are not automatically uploading to the hub

### ISSUE THREAD

**Issue by hamelsmu** [axolotl-ai-cloud/axolotl#1009]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/OpenAccess-AI-Collective/axolotl/labels/bug) didn't find any similar reports.

### Expected Behavior

When you set `hub_model_id` this should automatically upload the model to the hub.  I can see that the model README, etc is being put in `output_dir` but they are not being uploaded. 

I have logged in with `huggingface-cli login` as well as set `HUGGING_FACE_HUB_TOKEN` environment variable.

Some additional information:
- I created an **empty public** model repo on HuggingFace Hub
- Axolotl seems to create the repo for you properly if you don't already have one
- @winglian cannot reproduce this error :( 

### Current behaviour

Nothing is being uploaded to the Hub.  Before I dig into this issue more I would like to see if anyone else can repro this error.  The config I provided can be run in less than a minute

### Steps to reproduce

With the config, run `accelerate launch -m axolotl.cli.train config.yml`

### Config yaml

```yaml
base_model: openaccess-ai-collective/tiny-mistral
flash_attention: true
sequence_len: 1024
load_in_8bit: true
adapter: lora
lora_r: 32
lora_alpha: 64
lora_dropout: 0.05
lora_target_linear: true
val_set_size: 0.1
special_tokens:
  unk_token: <unk>
  bos_token: <s>
  eos_token: </s>
datasets:
  - path: mhenrichsen/alpaca_2k_test
    type: alpaca
num_epochs: 2
micro_batch_size: 2
gradient_accumulation_steps: 1
output_dir: temp_dir
learning_rate: 0.00001
optimizer: adamw_torch
lr_scheduler: cosine
max_steps: 20
save_steps: 10
eval_steps: 10
hub_model_id: hamel/axolotl-test-2
dataset_processes: 1
hf_use_auth_token: true
```

### Which Operating Systems are you using?

- [x] Linux
- [ ] macOS
- [ ] Windows

### Python Version

3.10.13

### axolotl branch-commit

main/384b817

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of axolotl.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

cc: @winglian

**Comment by NanoCode012** [comment]
Hello, I could not reproduce this as well. I only changed org to my personal HF. I did not create the public repo myself. Can you try with a new name but let transformers create the repo?

![image](https://github.com/OpenAccess-AI-Collective/axolotl/assets/9899957/fffc841d-cba1-40b1-abd3-2edb6b6cf817)

**Comment by hamelsmu** [comment]
Hmm ok so when I let axolotl create the repo it works, doesn't work if I already created repo.  I think this is ok.  I will add documentation.
